File: the-lambda-trilogy/typescript/lambdas/the-lambda-lith/lambdalith.ts
export {};

const express = require('express');

function apiRoutes(){
    const routes = new express.Router();

    routes.get('/add', (req:any, res:any) => {
        let firstNum = req?.query?.firstNum ?? 0;
        let secondNum = req?.query?.secondNum ?? 0;

        let result = Number(firstNum) + Number(secondNum);
        console.log(`result of ${firstNum} + ${secondNum} = ${result}`)

        res.status(200).json({"result":result})
    });

    routes.get('/subtract', (req:any, res:any) => {
        let firstNum = req?.query?.firstNum ?? 0;
        let secondNum = req?.query?.secondNum ?? 0;

        let result = Number(firstNum) - Number(secondNum);
        console.log(`result of ${firstNum} - ${secondNum} = ${result}`)

        res.status(200).json({"result":result})
    });

    routes.get('/multiply', (req:any, res:any) => {
        let firstNum = req?.query?.firstNum ?? 0;
        let secondNum = req?.query?.secondNum ?? 0;

        let result = Number(firstNum) * Number(secondNum);
        console.log(`result of ${firstNum} * ${secondNum} = ${result}`)

        res.status(200).json({"result":result})
    });

    return routes;
}


const app = express()
    .use(express.json())
    .use(apiRoutes());

const isInLambda = !!process.env.LAMBDA_TASK_ROOT;
if (isInLambda) {
    const serverlessExpress = require('aws-serverless-express');
    const server = serverlessExpress.createServer(app);
    exports.main = (event:any, context:any) => serverlessExpress.proxy(server, event, context)
} else {
    app.listen(3000, () => console.log(`Listening on 3000`));
}

File: the-lambda-trilogy/python/lambdas/the_lambda_lith/flask/lambdalith.py
import awsgi
from flask import (
    Flask,
    jsonify,
    request
)

app = Flask(__name__)


@app.route('/add')
def add():
    first_num = request.args.get('firstNum', default=0, type=int)
    second_num = request.args.get('secondNum', default=0, type=int)

    result = first_num + second_num
    print("The result of % s + % s = %s" % (first_num, second_num, result))
    return jsonify(result=result)


@app.route('/subtract')
def subtract():
    first_num = request.args.get('firstNum', default=0, type=int)
    second_num = request.args.get('secondNum', default=0, type=int)

    result = first_num - second_num
    print("The result of % s - % s = %s" % (first_num, second_num, result))
    return jsonify(result=result)


@app.route('/multiply')
def multiply():
    first_num = request.args.get('firstNum', default=0, type=int)
    second_num = request.args.get('secondNum', default=0, type=int)

    result = first_num * second_num
    print("The result of % s x % s = %s" % (first_num, second_num, result))
    return jsonify(result=result)


def handler(event, context):
    return awsgi.response(app, event, context, base64_content_types={"image/png"})
